### PR TITLE
mgr/dashboard: Add DimlessBinaryDirective

### PR DESCRIPTION
**Pull request by ricardoasmarques**
This PR adds a directive that can be used to provide a "size" input to the user.

screenshot from 2018-03-20 10-28-11

To use this directive, you can try the following code:

    <input cdDimlessBinary [(ngModel)]="model" [ngModelOptions]="{ updateOn: 'blur' }">

    <pre>{{ model }}</pre>

Signed-off-by: Ricardo Marques <rimarques@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/shared/directives/dimless-binary.directive.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { DimlessBinaryPipe } from '../pipes/dimless-binary.pipe';
+import { FormatterService } from '../services/formatter.service';
+
+@Directive({
+  selector: '[cdDimlessBinary]'
+})
+export class DimlessBinaryDirective implements OnInit {
+
+  @Output() ngModelChange: EventEmitter<any> = new EventEmitter();
+  @Input() minBytes: number;
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/directives/dimless-binary.directive.ts]: Can you please explain the input variables and how to use them, e.g. explain what values should/can be used.

Comment (ricardoasmarques) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/directives/dimless-binary.directive.ts]: I've added documentation for component inputs.

**Comment by votdev** [comment]
LGTM

File: src/pybind/mgr/dashboard/frontend/src/app/shared/services/formatter.service.ts [hunk 2]
```diff
@@ -97,4 +97,5 @@ export class FormatterService {
     }
     return bytes;
   }
+
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/services/formatter.service.ts]: You should remove this change.

Comment (ricardoasmarques) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/services/formatter.service.ts]: Good point. Done.

**Pull request merged**